ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx

```diff
@@ -117,15 +117,32 @@ export const PercentageAndAmountChange = ({
 
   const formattedValue = formatValue(balanceChange === 0 ? 0 : value, true);
 
-  const formattedValuePrice = isValidAmount(balanceChange)
-    ? `${(balanceChange as number) >= 0 ? '+' : ''}${Intl.NumberFormat(locale, {
-        notation: 'compact',
-        compactDisplay: 'short',
+  let formattedValuePrice = '';
+  if (isValidAmount(balanceChange)) {
+    formattedValuePrice = (balanceChange as number) >= 0 ? '+' : '';
+
+    const options = {
+      notation: 'compact',
+      compactDisplay: 'short',
+      maximumFractionDigits: 2,
+    } as const;
+
+    try {
+      // For currencies compliant with ISO 4217 Standard
+      formattedValuePrice += `${Intl.NumberFormat(locale, {
+        ...options,
         style: 'currency',
         currency: fiatCurrency,
-        maximumFractionDigits: 2,
-      }).format(balanceChange as number)} `
-    : '';
+      }).format(balanceChange as number)} `;
+    } catch {
+      // Non-standard Currency Codes
+      formattedValuePrice += `${Intl.NumberFormat(locale, {
+        ...options,
+        minimumFractionDigits: 2,
+        style: 'decimal',
+      }).format(balanceChange as number)} `;
```

Comment:
With standard currency codes, `Intl.NumberFormat` with `currency` argument would return the right currency formatted exp "$10" or "€5";

With non standard currency codes, we could do something like 
`.format(balanceChange as number)}  ${fiatCurrency}``; which would result in displaying the name of the currency;

I thought it is not necessary and we can display just the value for non standard currencies 

<img width="1132" alt="Screenshot 2024-09-18 at 11 11 38" src="https://github.com/user-attachments/assets/cbf4f09f-cf93-406d-b526-4bacc4c933c9">

Reply:
I think just the value is okay because the currency symbol is shown on the line above